.eslintrc.js

```diff
@@ -25,6 +25,13 @@ module.exports = {
 		'react/no-unused-prop-types': 'warn',
 		'prettier/prettier': 'warn',
 		'@typescript-eslint/explicit-module-boundary-types': 'off',
+		'@typescript-eslint/naming-convention': [
+			'error',
+			{
+				format: ['PascalCase'],
+				selector: ['typeLike'],
+			},
+		],
```

Comment:
Do you mind if I ask how eslint knows to only apply this rule to interfaces?

Reply:
So I read [this documentation](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md) to find the rule and associated values. If you scroll down to the selector area you can see the different selectors that are available. Below that, there are group selectors, which specifies the `typeLike` selector. This specifies `Interface`s `Class`es `Type`s and alike. I figured it made the most sense to apply this to all of those, and in doing so I only had to make one change to the whole repo, which means we are already following that or we haven't used some of those types yet.